docs/etcd-backup-restore-via-rke.md

```diff
@@ -0,0 +1,97 @@
+    ```
+    rke etcd snapshot-save --config cluster.yml --name <ETCD-SNAPSHOT-NAME>
+    ```
+    Example:
+    ```
+      $ rke etcd snapshot-save --config cluster.yml --name soil-2024-02-01T20:23:31Z_etcd
+        INFO[0000] Running RKE version: v1.3.10                 
+        INFO[0000] Starting saving snapshot on etcd hosts       
+        INFO[0000] [dialer] Setup tunnel for host [172.31.1.120]
+        INFO[0000] [dialer] Setup tunnel for host [172.31.3.195]
+        .....
+        .....
+        INFO[0037] [etcd] Successfully started [etcd-snapshot-once] container on host [172.31.1.120] 
+        INFO[0037] Waiting for [etcd-snapshot-once] container to exit on host [172.31.1.120]
+        INFO[0037] Container [etcd-snapshot-once] is still running on host [172.31.1.120]: stderr: [time="2024-02-01T14:56:38Z" level=info msg="Initializing Onetime Backup" name="soil-2024-02-01T20:23:31Z_etcd"
+        ], stdout: []
+        INFO[0038] Waiting for [etcd-snapshot-once] container to exit on host [172.31.1.120]
+        INFO[0039] Container [etcd-snapshot-once] is still running on host [172.31.1.120]: stderr: [time="2024-02-01T14:56:38Z" level=info msg="Initializing Onetime Backup" name="soil-2024-02-01T20:23:31Z_etcd"
+        ], stdout: []
+        INFO[0040] Waiting for [etcd-snapshot-once] container to exit on host [172.31.1.120]
+        INFO[0040] Removing container [etcd-snapshot-once] on host [172.31.1.120], try #1
+        INFO[0040] Finished saving/uploading snapshot [soil-2024-02-01T20:23:31Z_etcd] on all etcd hosts
+    ```
+    * The snapshot will be stored on all the nodes of k8s cluster under `/opt/rke/etcd-snapshots`. Login to one of the cluster nodes & verify.
+      ```
+      $ ssh -i /home/techno-384/Downloads/pem-keys/mosip-qa.pem ubuntu@172.31.1.12
+      ....
+      ....
+      ubuntu@ip-172-31-1-120:~$ cd /opt/rke/etcd-snapshots/
+      ubuntu@ip-172-31-1-120:/opt/rke/etcd-snapshots$ ls -lh
+      total 41M
+      -rw------- 1 root root 8.8M Jan 30 22:15 2024-01-30T22:15:19Z_etcd.zip
+      -rw------- 1 root root 8.5M Jan 31 10:15 2024-01-31T10:15:19Z_etcd.zip
+      -rw------- 1 root root 8.2M Jan 31 22:15 2024-01-31T22:15:19Z_etcd.zip
+      -rw------- 1 root root 7.9M Feb  1 10:15 2024-02-01T10:15:19Z_etcd.zip
+      -rw------- 1 root root 7.8M Feb  1 14:56 soil-2024-02-01T20:23:31Z_etcd.zip   ##### Here is the snapshot
+      ```
+
+* To enable recurring backup for the k8s cluster, update the `backup_config` section under `services` ---> `etcd` in `cluster.yml` and execute `rke up` command. 
+  ```
+  services:
+    etcd:
+      ....
+      ....
+      ....
+      backup_config:
+        interval_hours: 12
+        retention: 6
+  ```
+
+## Restore
+
+* To restore the state of the Kubernetes (k8s) cluster from a ETCD snapshot, execute the following command after updating the name of the snapshot (i.e., `<ETCD-SNAPSHOT-NAME>`).
```

Comment:
Why is ETCD capitalised in some places while it is etcd is the remaining? pls verify and update accordingly.